pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -186,3 +198,39 @@ func clusterOwnerRef(clusterName, clusterID string) metav1.OwnerReference {
 		BlockOwnerDeletion: &blockOwner,
 	}
 }
+
+func generateCrashEnvVar(v cephver.CephVersion) corev1.EnvVar {
+	val := fmt.Sprintf("-m $(ROOK_CEPH_MON_HOST) -k %s -n client.admin", keyring.VolumeMount().AdminKeyringFilePath())
+	if v.IsAtLeast(cephver.CephVersion{Major: 14, Minor: 2, Extra: 5}) {
+		val = fmt.Sprintf("-m $(ROOK_CEPH_MON_HOST) -k %s", keyring.VolumeMount().CrashCollectorKeyringFilePath())
+	}
+
+	env := corev1.EnvVar{Name: "CEPH_ARGS", Value: val}
+	return env
+}
+
+func getVolumeMounts(v cephver.CephVersion) corev1.VolumeMount {
+	volumeMounts := keyring.VolumeMount().Admin()
+
+	// As of Ceph Nautilus 14.2.5, the crash collector has its own key
+	// If not running on on at least this version let's use the Ceph admin key
+	// Thanks to https://github.com/ceph/ceph/pull/30844
+	if v.IsAtLeast(cephver.CephVersion{Major: 14, Minor: 2, Extra: 5}) {
+		volumeMounts = keyring.VolumeMount().CrashCollector()
+	}
+
+	return volumeMounts
+}
+
+func getVolumes(v cephver.CephVersion) corev1.Volume {
+	volumes := keyring.Volume().Admin()
```

Comment:
We're still always including the admin keyring in the pod? Again, seems like we should only add one of them to the pod and the pod picks it up in the same place.

Reply:
No, we override it right after when running on 14.2.5.